worlds/kdl3/Rules.py

```diff
@@ -0,0 +1,284 @@
+    return state.has("Rick", player) and state.has("Rick Spawn", player)
+
+
+def can_reach_kine(state: "CollectionState", player: int) -> bool:
+    return state.has("Kine", player) and state.has("Kine Spawn", player)
+
+
+def can_reach_coo(state: "CollectionState", player: int) -> bool:
+    return state.has("Coo", player) and state.has("Coo Spawn", player)
+
+
+def can_reach_nago(state: "CollectionState", player: int) -> bool:
+    return state.has("Nago", player) and state.has("Nago Spawn", player)
+
+
+def can_reach_chuchu(state: "CollectionState", player: int) -> bool:
+    return state.has("ChuChu", player) and state.has("ChuChu Spawn", player)
+
+
+def can_reach_pitch(state: "CollectionState", player: int) -> bool:
+    return state.has("Pitch", player) and state.has("Pitch Spawn", player)
+
+
+def can_reach_burning(state: "CollectionState", player: int) -> bool:
+    return state.has("Burning", player) and state.has("Burning Ability", player)
+
+
+def can_reach_stone(state: "CollectionState", player: int) -> bool:
+    return state.has("Stone", player) and state.has("Stone Ability", player)
+
+
+def can_reach_ice(state: "CollectionState", player: int) -> bool:
+    return state.has("Ice", player) and state.has("Ice Ability", player)
+
+
+def can_reach_needle(state: "CollectionState", player: int) -> bool:
+    return state.has("Needle", player) and state.has("Needle Ability", player)
+
+
+def can_reach_clean(state: "CollectionState", player: int) -> bool:
+    return state.has("Clean", player) and state.has("Clean Ability", player)
+
+
+def can_reach_parasol(state: "CollectionState", player: int) -> bool:
+    return state.has("Parasol", player) and state.has("Parasol Ability", player)
+
+
+def can_reach_spark(state: "CollectionState", player: int) -> bool:
+    return state.has("Spark", player) and state.has("Spark Ability", player)
+
+
+def can_reach_cutter(state: "CollectionState", player: int) -> bool:
+    return state.has("Cutter", player) and state.has("Cutter Ability", player)
+
+
+ability_map = {
+    "No Ability": lambda state, player: True,
+    "Burning Ability": can_reach_burning,
+    "Stone Ability": can_reach_stone,
+    "Ice Ability": can_reach_ice,
+    "Needle Ability": can_reach_needle,
+    "Clean Ability": can_reach_clean,
+    "Parasol Ability": can_reach_parasol,
+    "Spark Ability": can_reach_spark,
+    "Cutter Ability": can_reach_cutter,
+}
+
+
+def can_assemble_rob(state: "CollectionState", player: int, copy_abilities: typing.Dict[str,str]):
+    # check animal requirements
+    if not can_reach_coo(state, player) and can_reach_kine(state, player):
+        return False
+    # probably some cleaner way to handle this
+    room1 = EnemyAbilities.enemy_restrictive[1]
+    room2 = EnemyAbilities.enemy_restrictive[2]
+    room3 = EnemyAbilities.enemy_restrictive[3]
+    room4 = EnemyAbilities.enemy_restrictive[4]
+    for abilities, bukisets in [room1, room2, room3, room4]:
```

Comment:
Could simplify by removing the previous four lines and change this to:
```suggestion
    for abilities, bukisets in EnemyAbilities.enemy_restrictive[1:5]
```